MSAL/src/cache/MSALTokenCache.m

```diff
@@ -158,11 +178,11 @@ - (MSALAccessTokenCacheItem *)findAccessToken:(MSALRequestParameters *)requestPa
     
     for (MSALAccessTokenCacheItem *tokenItem in allAccessTokens)
     {
-        if (requestParam.unvalidatedAuthority && [key matches:[tokenItem tokenCacheKey:nil]])
+        if (authority && [key matches:[tokenItem tokenCacheKey:nil]])
```

Comment:
I've moved things around, no more need for MSALAccessTokenCacheKey and [key matches:]

Things checked:
- scopes: checked as a first condition to match
- clientId: filtered when allAccessTokensForUser: is called
- authority: checked after scopes.
- user: filtered when allAccessTokensForUser: is called